lib/chef-acceptance/application.rb

```diff
@@ -4,62 +4,91 @@
 require "chef-acceptance/executable_helper"
 
 module ChefAcceptance
-  class AcceptanceError < StandardError; end
+  class AcceptanceFatalError < StandardError; end
 
   # Responsible for running one or more suites and
   # displaying statistics about the runs to the user.
   class Application
     attr_reader :force_destroy
     attr_reader :output_formatter
+    attr_reader :errors
 
     def initialize(options = {})
       @force_destroy = options.fetch("force_destroy", false)
       @output_formatter = OutputFormatter.new
+      @errors = {}
     end
 
-    def run(suite, command)
-      start_time = Time.now
-      error = false
-      run_summary = true
-
+    def run(suite_regex, command)
       begin
-        run_suite(suite, command)
-      rescue AcceptanceError => e
-        error = true
-        run_summary = false
-        puts "#{e}\n#{e.backtrace.join("\n")}"
-      rescue RuntimeError
-        # We catch the errors here and do not raise again in
-        # order to make a clean exit. Since the errors are already
-        # in stdout, we do not print them again.
-        error = true
-      ensure
-        if run_summary
-          total_duration = Time.now - start_time
-          # Special footer row that gives overall status
-          output_formatter.add_row(suite: "", command: "Total", duration: total_duration, error: error)
-
-          puts ""
-          puts "chef-acceptance run #{error ? "failed" : "succeeded"}."
-          puts output_formatter.generate_output
+        suites = parse_suites(suite_regex)
+
+        puts "Running test suites:"
+        puts suites.join("\n")
+        puts ""
+
+        # Start the overall run timer
+        run_start_time = Time.now
+
+        suites.each do |suite|
+          suite_start_time = Time.now
+
+          begin
+            run_suite(suite, command)
+          rescue RuntimeError => e
+            # We catch the errors here and do not raise again in
+            # order to make a clean exit. Since the errors are already
+            # in stdout, we do not print them again.
+            errors[suite] = e
+          ensure
+            suite_duration = Time.now - suite_start_time
+            # Capture overall suite run duration
+            output_formatter.add_row(suite: suite, command: "Total", duration: suite_duration, error: errors[suite])
+          end
         end
+
+        total_duration = Time.now - run_start_time
+        # Special footer row that gives overall status for run
+        output_formatter.add_row(suite: "Run", command: "Total", duration: total_duration, error: run_error?)
+        print_summary
+      rescue AcceptanceFatalError => e
```

Comment:
`run_suites` has the potential to raise `AcceptanceFatalError` also I think this structure is good in the sense that anywhere we raise an `AcceptanceFatalError` we have a healthy clean exit. I would vote to keep it this way.